elastic/kibana
# [Infra UI] Add custom threshold alerts for serverless

Closes https://github.com/elastic/kibana/issues/166612

## Summary

- Adds a `customThresholdAlertsEnabled` feature flag
- Based on the feature flag shows an additional menu item in the alerts dropdown
- Adds flyout component for rendering the custom alerts flyout

**Serverless**
<video src="https://github.com/elastic/kibana/assets/793851/4fc79427-1e0c-4692-8525-b3b35e8a04e9" controls="controls"></video>

**Traditional**
![Traditional](https://github.com/elastic/kibana/assets/793851/b2c16478-a40d-4312-a41e-5e34c98daa47)

## How to test

- Checkout locally and run in serverless mode (infra plugin on)
- Make sure you see the "Create custom threshold alert" menu item in the "Alerts and rules" dropdown
- Click on the item and make sure you can create an alert
- Run Kibana in traditional mode and make sure the dropdown item is not there
